### ISSUE TITLE
(Question) How did you customize Dolphin to make phillip work?

### ISSUE THREAD

**Issue by neobis001** [vladfi1/phillip#27]
Hi, I'm trying to learn more about the changes you made to the emulator source code. I want to study the emulator code itself more. It may help me with my own custom AI. I'm hoping to discuss your changes for a while, so I can try to learn a lot from it.

**Comment by vladfi1** [comment]
If you're interested in melee AI I wouldn't bother with my [dolphin](http://github.com/vladfi1/dolphin-emu), and instead use altf4's [libmelee](https://github.com/altf4/libmelee).

**Comment by neobis001** [comment]
No, it's for another game, Soul Calibur 2. I modified my Dolphin code with this "MemoryWatcher" and "Pipes" thing. I learned about it looking at altf4's repos, spxtr's, and also yours. I don't have an AI now, but I'm feeling confident with being able to read memory and send controller inputs.

-I'd like to continue it by learning more about Dolphin's source code. I checked your commits at https://github.com/vladfi1/dolphin/commits/master. It looks like you understand Dolphin well. Do you have advice for understanding Dolphin's source, or do you know anybody that understands Dolphin well? It's the emulator I want to discuss.

-By the way, you're the person who did the "deep learning" thing with smash, right? This one (https://arxiv.org/abs/1702.06230). The math is too complicated for me, but I'm still curious about it. If I wanted to ask you about it, is there any way I can contact you?

**Comment by vladfi1** [comment]
- I know Dolphin a bit, but if you want help with it I'd recommend going to their IRC channel (`dolphin-emu` on Freenode).
- Yes, this is the repo for deep learning melee. I don't have much bandwidth for answering questions, but if you're interested in learning on your own I can recommend David Silver's lectures on Youtube as well as [the book](http://incompleteideas.net/book/the-book.html) on Reinforcement Learning.
